controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -114,6 +114,18 @@ func AkoAddonSecretDataYaml(cluster *clusterv1.Cluster, obj *akoov1alpha1.AKODep
 	if err != nil {
 		return "", err
 	}
+
+	//Avoid setting DeleteConfig for management cluster
+	if cluster.Namespace != akoov1alpha1.TKGSystemNamespace {
+		if deleteConfig, exists := cluster.Labels[akoov1alpha1.AviClusterDeleteConfigLabel]; exists {
+			if deleteConfig == "true" {
+				secret.LoadBalancerAndIngressService.Config.AKOSettings.DeleteConfig = "true"
+			}
+		} else {
```

Comment:
The default value is "false", we can safely delete this `else` condition.

Reply:
This happens when customer removed the label after he set it to true

Reply:
Looks like ako.NewValues() will initialize deleteConfig to false. So, else case is not required

Reply:
Yes, it is not required, the add-on secret will be re-generated, so it will be `false`